File: php/src/Framework/SecurityUser/FrameworkSecurityService.php
<?php
declare(strict_types=1);

namespace Demo\App\Framework\SecurityUser;

use Demo\App\Framework\FrameworkRequest;

final class FrameworkSecurityService
{
    public function __construct(private SecurityUserRepository $securityUserRepository)
    {
    }

    public function getSecurityUserFromRequest(FrameworkRequest $request): ?SecurityUser
    {
        return $this->securityUserRepository->findUserById($request->headers()['userSession']);
    }
}

File: kotlin/src/main/kotlin/framework/securityuser/FrameworkSecurityService.kt
package framework.securityuser

import framework.FrameworkRequest

class FrameworkSecurityService(private val securityUserRepository: SecurityUserRepository) {

    fun getSecurityUserFromRequest(request: FrameworkRequest): SecurityUser? {
        return securityUserRepository.findUserById(request.headers["userSession"]!!)
    }
}
